esp/esp/program/modules/handlers/onsiteclasslist.py

```diff
@@ -188,7 +188,8 @@ def register_student(self, request, tl, one, two, module, extra, prog):
     @needs_onsite
     def checkin_status(self, request, tl, one, two, module, extra, prog):
         resp = HttpResponse(content_type='application/json')
-        data = ESPUser.objects.filter(record__event="attended", record__program=prog).distinct().values_list('id')
+        students = prog.students(True)
+        data = ESPUser.objects.filter(students['attended'] & ~students['checked_out']).distinct().values_list('id')
```

Comment:
It doesn't seem to be filtering out checked out students from the counts of students checked in on grid based class changes for me...

Reply:
This seems related to your next comment.